common/src/main/java/com/itsaky/androidide/utils/AndroidUtils.java

```diff
@@ -0,0 +1,203 @@
+package com.itsaky.androidide.utils;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.*;
+
+public class AndroidUtils {
+  /**
+   * The package is used to create a directory (eg:
+   * MyApplication/app/src/main/java/src/my/package/name) A windows directory path cannot be longer
+   * than 250 chars On unix/mac a directory name cannot be longer than 250 chars On all platforms,
+   * aapt fails with really cryptic errors if the package name is longer that ~200 chars Having a
+   * sane length for the package also seems a good thing
+   */
+  private static final int PACKAGE_LENGTH_LIMIT = 100;
+
+  /**
+   * Checks if the given name is a valid Android application package (which has additional
+   * requirements beyond a normal Java package)
+   *
+   * @see #validateAndroidPackageName(String)
+   */
+  public static boolean isValidAndroidPackageName(@NotNull String name) {
+    return validateAndroidPackageName(name) == null;
+  }
+
+  /**
+   * Checks if the given name is a valid general Java package name.
+   *
+   * <p>If validating the Android package name, use {@link #validateAndroidPackageName(String)}
+   * instead!
+   */
+  public static boolean isValidJavaPackageName(@NotNull String name) {
+    int index = 0;
+    while (true) {
+      int index1 = name.indexOf('.', index);
+      if (index1 < 0) index1 = name.length();
+      if (!isIdentifier(name.substring(index, index1))) return false;
+      if (index1 == name.length()) return true;
+      index = index1 + 1;
+    }
+  }
+
+  /**
+   * Validates a potential package name and returns null if the package name is valid, and otherwise
+   * returns a description for why it is not valid.
+   *
+   * <p>Note that Android package names are more restrictive than general Java package names; we
+   * require at least two segments, limit the character set to [a-zA-Z0-9_] (Java allows any {@link
+   * Character#isLetter(char)} and require that each segment start with a letter (Java allows
+   * underscores at the beginning).
+   *
+   * <p>For details, see core/java/android/content/pm/PackageParser.java#validateName
+   *
+   * @param name the package name
+   * @return null if the package is valid as an Android package name, and otherwise a description
+   *     for why not
+   */
+  @Nullable
+  public static String validateAndroidPackageName(@NotNull String name) {
+    if (name.isEmpty()) {
+      return "Package name is missing";
```

Comment:
Most of the strings in this class are shown to the user. Please extract them to `strings.xml`.